features/netsocket/UDPSocket.h

```diff
@@ -99,14 +99,14 @@ class UDPSocket : public Socket {
      *  Receives data and stores the source address in address if address
      *  is not NULL. Returns the number of bytes received into the buffer.
      *
-     *  By default, recvfrom blocks until data is sent. If socket is set to
+     *  By default, recvfrom blocks until data is received. If socket is set to
      *  non-blocking or times out, NSAPI_ERROR_WOULD_BLOCK is returned
      *  immediately.
      *
      *  @param address  Destination for the source address or NULL
      *  @param data     Destination buffer for data received from the host
      *  @param size     Size of the buffer in bytes
-     *  @return         Number of received bytes on success, negative error
+     *  @return         Size of received datagram, negative error
```

Comment:
Unfortunately, this wording is wrong in a different way - if the datagram is bigger than your buffer, it should return the bytes actually read - it's quietly truncated. (Which means applications have to provide known-to-be-big-enough buffers for whatever protocol they're implementing).

The body is too wooly about "receiving data", but at least the brief is already correct, I see - it does say "Receive a packet", compared to "Receive data" on `TCPSocket::recv`.

How about transforming to "datagram" everywhere:

    /** Receive a datagram over a UDP socket
     *
     *  Receives a datagram and stores the source address in address if address
     *  is not NULL. Returns the number of bytes written into the buffer. If the
     *  datagram is larger than the buffer, the excess data is silently discarded.
     *
     *  By default, recvfrom blocks until a datagram is received. If socket is set to
     *  non-blocking or times out with no datagram, NSAPI_ERROR_WOULD_BLOCK
     *  is returned immediately.
     *
     *  @param address  Destination for the source address or NULL
     *  @param data     Destination buffer for datagram received from the host
     *  @param size     Size of the buffer in bytes
     *  @return         Number of received bytes on success, negative error
     *                  code on failure
     */